Микросетки на флексах для всех страниц

DianaLad8 opened the pull request:
---
:mortar_board: [Микросетки на флексах для всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2189703/tasks/12)

:boom: https://htmlacademy-htmlcss.github.io/2189703-sedona-36/6/

DianaLad8 committed:
Добавляет микросетку на флексах на главную страницу

DianaLad8 committed:
Выравнивает интерактивные элементы в подвале

Увеличивает площадь клика

DianaLad8 committed:
Стилизует шапку страницы каталога и фильтрацию

Результаты фильтрации - стилизует строку с поиском и кнопками вида просмотра. Карточки отфильтрованных отелей оставила на гридов.

DianaLad8 committed:
Стилизует пагинацию

DianaLad8 committed:
Добавляет вертикальные отступы между крупными блокамид

DianaLad8 committed:
Добавляет вложенный список в статью-описание

kamunicorn commented on styles/styles.css:
```diff
@@ -142,15 +176,25 @@ body {
+  padding: 51px 0 82px;
   background-color: #3F5E72;
   background-image: url(../images/hero-background.jpg);
   background-size: 100% auto;
```
если задать `background-size: cover`, то фон растянется на размеры блока
отключить повторение фона
![image](https://user-images.githubusercontent.com/38280676/218597855-58c15c15-37cc-4af0-b81a-7d2d5548b649.png)

kamunicorn commented on styles/styles.css:
```diff
@@ -142,15 +176,25 @@ body {
-  background-color: #ffffff;
+
+.description-text.heading {
+  padding: 69px 290px 15px 290px;
```
1. Вертикальные отступы между элементами должны задаваться через `margin`.
2. Горизонтальные отступы этим элементам не нужно задавать. По крайней мере, такие, чтоб было впритык по макету. Лучше добавить небольшие отступы, чтобы большой абзац не превращался в длинный лоскут.
Вот как будет выглядеть этот блок, если добавить еще абзац и текст:
![image](https://user-images.githubusercontent.com/38280676/218599146-b4dc0bf7-ab6a-46f9-9578-64783a0b91fa.png)

DianaLad8 replied:
> 1. Вертикальные отступы между элементами должны задаваться через `margin`.
> 2. Горизонтальные отступы этим элементам не нужно задавать. По крайней мере, такие, чтоб было впритык по макету. Лучше добавить небольшие отступы, чтобы большой абзац не превращался в длинный лоскут.
>    Вот как будет выглядеть этот блок, если добавить еще абзац и текст:
>    ![image](https://user-images.githubusercontent.com/38280676/218599146-b4dc0bf7-ab6a-46f9-9578-64783a0b91fa.png)

Здесь как по лайву в понедельник сделала...

kamunicorn replied:
Хорошо получилось (переделанное)👍🏻

kamunicorn commented on styles/styles.css:
```diff
@@ -159,23 +203,45 @@ body {
+}
+
+.item-no-images .description-title {
+  margin-top: 112px;
```
![image](https://user-images.githubusercontent.com/38280676/218599585-1cb574c2-a6be-4fef-a6f8-415a6466ec7f.png)

kamunicorn commented on styles/styles.css:
```diff
@@ -187,28 +253,53 @@ body {
+}
+
+.search-title {
+  margin-bottom: 20px;
```
Вот тут хорошо - паддингов нет, отступы за счет маржинов.

DianaLad8 replied:
Этот раздел полностью переделала

kamunicorn commented on styles/styles.css:
```diff
@@ -187,28 +253,53 @@ body {
+
 .search {
-  background-color: #ffffff;
+  padding: 96px 304px;
```
Вот тут как раз вертикальные отступы секции за счет внутренних отступов секции, а не текста внутри.
Горизонтальный отступ не обязательно делать такой впритык. 
Переносить строки можно за счет `<br>`. Чтобы не висели предлоги и мелкие слова, можно использовать неразрывный пробел `&nbsp;`.

kamunicorn commented on styles/styles.css:
```diff
@@ -217,11 +308,28 @@ body {
 }
 
+.subscribe-title {
+  padding: 96px 398px 10px 398px;
```
![image](https://user-images.githubusercontent.com/38280676/218600631-f11cb952-4b6b-4885-b1a0-ff24bcde26e3.png)
Потому что вертикальное пространство блока задается через отступ дочернего элемента. Если заголовок убрать, сломается отступ всей секции, будет не эстетично

kamunicorn commented on styles/styles.css:
```diff
@@ -417,11 +585,7 @@ border: none;
+/* Звездность и рейтинг отеля */
 .hotel-four-star {
   background-image: url(../images/star.svg);
   background-repeat: repeat-x;
```
На отлично справилась со сложным элементом🔥
Осталось добавить классы для остальных звезд

kamunicorn commented on styles/styles.css:
```diff
@@ -444,36 +608,54 @@ border: none;
   display: flex;
   width: 1060px;
-  margin: 0 auto;
+  margin: 60px 70px 80px;
```
Я бы предложила сделать еще один стакан для горизонтальных отступов - везде 1060px. Можно просто классом или дивом, но див могут посчитать лишним элементом.

kamunicorn commented on styles/styles.css:
```diff
@@ -444,36 +608,54 @@ border: none;
+/* Пагинация */
+
+.pagination-list {
   display: flex;
```
![image](https://user-images.githubusercontent.com/38280676/218610470-45e6d8d1-c4ec-41ef-b1f3-a273eb365aa1.png)
Нет `flex-wrap: wrap`, поэтому не проходит проверку переполнением.

kamunicorn commented on styles/styles.css:
```diff
@@ -236,58 +344,91 @@ body {
+  display: block;
+  box-sizing: border-box;
+  padding: 50px 12px;
+  height: 120px;
```
![image](https://user-images.githubusercontent.com/38280676/218610737-fc0120b0-61d6-4f5c-9783-25322f32d25d.png)
Проходит проверку переполнением, но есть 2 проблемы:
1. При переходе ссылок на второй ряд они слишком отдаляются друг от друга, потому что им задана большая высота вместо того, чтобы задать отступы у футера .
2. сильно ограниченна ширина, хотя список мог бы дальше продолжаться и это выглядело нормально.

kamunicorn commented on styles/styles.css:
```diff
@@ -217,11 +308,28 @@ body {
+ }
+
 .button-subscribe {
+
```
Этой и другим кнопкам уже можно задавать внутренние отступы

kamunicorn commented on styles/styles.css:
```diff
@@ -374,40 +518,64 @@ border: none;
+/* Карточка отеля */
+
+.hotel-title {
+  display: inline-block;
```
поэтому заголовок слипается с текстом "Гостиница"
![image](https://user-images.githubusercontent.com/38280676/218612041-a39a8eb8-61ce-4302-94ae-be2b4227ea05.png)

kamunicorn commented on styles/styles.css:
```diff
@@ -236,58 +344,91 @@ body {
+  text-align: left;
+  background-color: #3F5E72;
+  background-image: url(../images/filter-background.jpg);
+  background-repeat: no-repeat;
```
![image](https://user-images.githubusercontent.com/38280676/218612958-d6be5e83-9912-440e-adde-3e3c685f1cc8.png)
Повтора нет, фоновый цвет есть 👍🏻
Изображение не растягивается на высоту блока.

kamunicorn reviewed (commented):
- Стандартизировать вертикальные отступы для текста абзацев: переделать из внутренних во внешние (для элемента или родителя, секции). В некоторых секциях так уже сделано👍🏻
- Исправить ошибки с фоновыми картинками.
- Добавить переполнение блоков (пагинация).
- Убрать задание ширины впритык, добавить "простора": в иконках соцсетей, в горизонтальных отступах.
- Добавить внутренние отступы кнопкам. Для некоторых кнопок так уже сделано👍🏻

kamunicorn commented on catalog.html:
```diff
@@ -6,265 +6,267 @@
-            <li class="breadcrumbs-item">
-              <a class="breadcrumbs-link">Гостиницы</a>
+            <li class="navigaton-item">
+              <a class="navigation-link button" href="#">Хочу сюда!</a>
```
В верстке эта кнопка получилась больше, чем должна, потому что её не задан `box-sizing: border-box`. По умолчанию, элементы имеют `box-sizing: content-box`.
В [статье про отступы](https://up.htmlacademy.ru/htmlcss/36/book/css/margin-padding) подробнее написано про box-sizing. Во в остальных стилях я вижу, что ты его уже применяешь👍🏻

DianaLad8 replied:
Кнопки все переделала)) В начале стилевого файла общие правила поменяла и точечно по разметке, где различия))

kamunicorn commented on index.html:
```diff
@@ -6,156 +6,169 @@
+              <a class="navigation-link" href="#">О Cедоне</a>
+            </li>
+            <li class="navigaton-item">
+              <a class="navigation-link" href="catalog.html">Гостиницы</a>
```
Вижу, что подчеркивание сделано с помощью фоновой картинки:)
В разделе 8 про декоративные элементы будут показывать, как делаются такие вещи. Конкретно этот элемент можно сделать разные способами.
![image](https://user-images.githubusercontent.com/38280676/218614742-66268875-e94e-42ff-9362-44267965527f.png)

DianaLad8 replied:
Еле доперла, откуда эта линия :))

kamunicorn reviewed (commented):
Замечания:
- по кнопке Хочу сюда в навигации,
- и фоновой картинке в каталоге.

DianaLad8 committed:
Исправляет внешние и внутренние отступы, меняет правила для фона

Текстовые блоки обернула в отдельные блоки и добавила отступы к кнопкам, как показали на лайве.
Фон в каталоге заменила, правила для фонов изменила на "cover".
Карточки отелей оставила для следующего ДЗ.

DianaLad8 committed:
Еще раз исправляет пагинацию

После массового одобрения в Дискорте))

DianaLad8 committed:
Меняет разметку и стили к статье-описанию другим способом

Автор идеи - Дмитрий Халимов

DianaLad8 commented:
Практикуюсь комментировать в гитхабе)

kamunicorn commented on styles/styles.css:
```diff
@@ -142,15 +183,26 @@ body {
+.description-wrapper {
+  width: 800px;
+  margin: 0 auto;
+  margin-bottom: 21px;
```
Лишний маржин, все можно сделать через padding
![image](https://user-images.githubusercontent.com/38280676/219257781-c44cf7c6-5f85-462b-b9b3-25bd8e76a0cf.png)
Я поняла, что это в макете так - секция одна и еще расстояние до следующего списка. Это просто косяк в дизайне, надо измерять настоящее расстояние от текста до секции:
![image](https://user-images.githubusercontent.com/38280676/219258136-6f979ffe-3c20-446a-a6e0-2f700be85718.png)

kamunicorn commented on styles/styles.css:
```diff
@@ -187,41 +258,100 @@ body {
 }
 
+ .subscribe-wrapper {
+  width: 800px;
```
Если отключить ширину, ничего не произойдет. Значит свойство лишнее.

kamunicorn commented on index.html:
```diff
@@ -6,156 +6,183 @@
+        </section>
+
+        <section class="subscribe" data-test="subscribe">
+          <!-- После рестроспективы добавила обертку к тексту -->
```
Лишний элемент, можно обойтись без него. Если на лайве говорили об этом, напишите, на какой минуте, разберем, может не то имелось ввиду.

kamunicorn commented on styles/styles.css:
```diff
@@ -187,41 +258,100 @@ body {
+ .subscribe-wrapper {
+  width: 800px;
+  margin: 0 auto;
+  padding-top: 96px;
```
Лучше его перенести в отступ секции

kamunicorn commented on styles/styles.css:
```diff
@@ -187,41 +258,100 @@ body {
+  width: 800px;
+  margin: 0 auto;
+  padding-top: 96px;
+  margin-bottom: 54px;
```
можно перенести в верхний отступ формы

kamunicorn commented on styles/styles.css:
```diff
@@ -233,61 +363,94 @@ body {
+  background-color: #3F5E72;
+  background-image: url(../images/filter-background.jpg);
+  background-size: cover;
+  color: #ffffff;
```
Лучше будет задать этому элементу внутренние отступы:
![image](https://user-images.githubusercontent.com/38280676/219261526-8e594bec-9b60-4262-8a10-a1a9046758bc.png)
Вместо того, чтобы задавать их его детям `.header-inner` (горизонтальные), `.inner-header-title` (верхний отступ), `.catalog-filter-hotels` (горизонтальные), `.catalog-filter` (нижний отступ). Будет порядок и удобнее менять: сразу в одном месте, а не в четырех местах, легко запутаться.

kamunicorn commented on styles/styles.css:
```diff
@@ -303,66 +466,73 @@ body {
 
 .catalog-filter-group {
 width: 150px;
```
Убрать жесткую ширину.
Можно расположить эти фильтры и кнопки за счет маржинов, прижать к правому краю за счет `margin-right: 0`, `margin-left: auto` и т.п.

kamunicorn reviewed (commented):
Замечания по отступам блока с подпиской и фильтра гостиниц, расположению фильтров внутри формы.
Переполнение и отступы абзацев исправлены :thumbsup:
